### PR TITLE
Allow use some additional config for old sqlserver

### LINKED ISSUE #16569: Please add some custom option for Sqlserver driver
### Description

Currently, there are no way to pass some options for obdc driver.
 So, i have to modified Sqlserver to provide that support

Currently i had to modify Database/Driver/Sqlserver.php

```
        if ($config['encrypt'] !== null) {
            $dsn .= ";Encrypt={$config['encrypt']}";
        }
        if ($config['trustServerCertificate'] !== null) {
            $dsn .= ";TrustServerCertificate={$config['trustServerCertificate']}";
        }
```

and the config file looks like:

```
       'default' => [
            'className' => 'Cake\Database\Connection',
            'driver' => 'Cake\Database\Driver\Sqlserver',
            'persistent' => false,
            'host' => getEnvOrFail('MSSQL_HOST'),
            'username' => getEnvOrFail('MSSQL_USER'),
            'password' => getEnvOrFail('MSSQL_PASSWORD'),
            'database' => getEnvOrFail('MSSQL_DATABASE'),
            'flags' => [
                \PDO::ATTR_ERRMODE => false
            ],
            'cacheMetadata' => true,
            'log' => false,
            **'encrypt' => 'no',
            'trustServerCertificate' => 'yes'**
        ],
```

Whitout thes options obdc18 does not work in old mssql servers

### CakePHP Version

4.x

### PR DESCRIPTION
**Pull request by CriztianiX**
Allow use some additional config for old sqlserver with obdc18

### CODE REVIEW

File: src/Database/Driver/Sqlserver.php [hunk 1]
```diff
@@ -151,6 +151,12 @@ public function connect(): bool
         if ($config['multiSubnetFailover'] !== null) {
             $dsn .= ";MultiSubnetFailover={$config['multiSubnetFailover']}";
         }
+        if ($config['encrypt'] !== null) {
```
Comment (othercorey) [inline src/Database/Driver/Sqlserver.php:156]: These must be added to `$_baseConfig` above.

Comment (CriztianiX) [reply src/Database/Driver/Sqlserver.php:156]: done

**Comment by othercorey** [comment]
The windows test failures need to be fixed.

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**